src/apps/HeyGPT/HeyGPT/App.xaml.cs

```diff
@@ -20,17 +19,79 @@
 
 namespace HeyGPT.App;
 
-// To learn more about WinUI 3, see https://docs.microsoft.com/windows/apps/winui/winui3/.
 public partial class App : Application
 {
-    // The .NET Default Host provides dependency injection, configuration, logging, and other services.
-    // https://docs.microsoft.com/dotnet/core/extensions/generic-host
-    // https://docs.microsoft.com/dotnet/core/extensions/dependency-injection
-    // https://docs.microsoft.com/dotnet/core/extensions/configuration
-    // https://docs.microsoft.com/dotnet/core/extensions/logging
-    public IHost Host
+    public IHost Host { get; }
+
+    // TODO: We probably want to manage the MainWindow in a dedicated service.
+    // Currently, anybody that needs access to the MainWindow will need to use the
+    // App global object. We also may want to handle showing/hiding the main window
+    // as well as multiple child windows.  And we may want to respond on events on the
+    // MainWindow.
+    // Currently, there is some coupling with the AppTitleBar helpers and other places
+    // where global access to the App object is being used.  Those need to be settled first.
+    public static WindowEx MainWindow { get; } = new MainWindow();
+
+    public static UIElement? AppTitlebar { get; set; }
+
+    public App()
     {
-        get;
+        InitializeComponent();
+
+        Host = Microsoft.Extensions.Hosting.Host.CreateDefaultBuilder().UseContentRoot(AppContext.BaseDirectory)
+            .ConfigureServices((context, services) =>
+            {
+                // Activation Handler(s)
+
+                // Default Activation Handler
+                services.AddKeyedTransient<IActivationHandler, DefaultActivationHandler>(ActivationHandlerKeys.DefaultServiceKey);
```

Comment:
Here we are registering our different activation handlers.   We use Keyed Service registration to support injecting the default from the Enumerated list containing the others.